#159987730 Users should be able to like or dislike comment made on an article.

emekafredy opened the pull request:
# What does this PR do?
Create endpoints for to like or dislike article comments

# Description of Task to be completed?
### Create CommentLike model and migration file
### Write unit tests for feature
### Create endpoint
- Like comment ``` POST api/articles/<article_slug>/comments/<commentId>/like ```
- Dislike comment ``` POST api/articles/<article_slug>/comments/<commentId>/dislike ```

# How should this be manually tested?
- Clone the repo
- cd into the 'Thor-ah'
- Run 'npm install'
- Use the .env.sample format to create a .env file
- Run 'npm run migrate'
- Run 'npm run start:dev'

# Any background context you want to provide?
N/A

# What are the relevant pivotal tracker stories?
[159987730](https://www.pivotaltracker.com/story/show/159987730)

emekafredy committed:
feat(commentLikesDislikes): endpoint to like or dislike comments on articles

- define CommentLikes Model and its association with comments and users.
- write unit tests for comment like model and route.
- create endpoints for authenticated users to like or dislike comments made on articles.

[Delivers #159987730]

CEOehis reviewed (approved):
LGTM

lauragift21 reviewed (commented):
LGTM

lauragift21 reviewed (changes requested):
For naming your models, you can add underscore between the words making it comment_like.js.

lauragift21 reviewed (approved):
LGTM

emekafredy committed:
feat(commentLikesDislikes): endpoint to like or dislike comments on articles

- redefine model to store integers (0, 1) for dislikes and likes
- write endpoint to get all reactions of a comment

[Delivers #159987730]

emekafredy commented:
@lauragift21 This has been resolved

aknwosu commented on server/controllers/comments.js:
```diff
@@ -122,6 +122,204 @@ class CommentsController {
+  }
+
+  /**
+   * @description get number of likes and dislikes in an article
```
misleading comment

emekafredy replied:
Thanks. I will correct this

aknwosu commented on server/controllers/comments.js:
```diff
@@ -122,6 +122,204 @@ class CommentsController {
+              if (commentReaction.reaction === '1') {
+                commentReactions.likes += 1;
+              } else if (commentReaction.reaction === '0') {
+                commentReactions.dislikes += 1;
```
a better implementation is to find all where reaction === 0 or reaction === 1 and then total number of likes would be the length of the returned value.

emekafredy replied:
Thank you. I will also work on this

aknwosu reviewed (changes requested):
add a function to check if the current logged in user already liked or disliked a comment so it can be used to format the view in the front end

emekafredy committed:
feat(commentLikesDislikes): endpoint to like or dislike comments on articles

- Refactor code on getting the number of likes and dislikes in a comment.
- Define an endpoint to check if the current logged in user already liked or disliked a comment.

[Delivers #159987730]
